packages/client/src/schema/repository.ts

```diff
@@ -182,24 +189,36 @@ export class RestRepository<Data extends BaseData, Record extends XataRecord = D
   ): Promise<SelectedPick<Record, ['*']> | SelectedPick<Record, ['*']>[]> {
     // Create many records
     if (Array.isArray(a)) {
-      return this.#bulkInsertTableRecords(a);
+      const records = await this.#bulkInsertTableRecords(a);
+      await Promise.all(records.map((record) => this.recordCache.set(`${this.#table}-${record.id}`, record)));
```

Comment:
We still have potential clashing here, because `-` is an allowed character for tables.

So if you have table=foo-bar key=baz, and table=foo key=bar-baz, they clash.

Instead of `-` we can use `:` or something else.